### PR TITLE
refactor: use createSlice for network reducer

### PR DESCRIPTION
**Pull request by vshakun**
Part of #799 

Use `createSlice` for network reducer according to [topic #1 proposal #3](https://github.com/spacemeshos/smapp/issues/799#p1).

### CODE REVIEW

File: app/redux/network/actions.ts [hunk 1]
```diff
@@ -1,25 +1,22 @@
 import { eventsService } from '../../infra/eventsService';
 import { AppThDispatch } from '../../types';
-
-export const SET_NETWORK_DEFINITIONS = 'SET_NETWORK_DEFINITIONS';
-export const SET_CURRENT_LAYER = 'SET_CURRENT_LAYER';
-export const SET_STATE_ROOT_HASH = 'SET_STATE_ROOT_HASH';
+import slice from './slice';
 
 export const getNetworkDefinitions = () => async (dispatch: AppThDispatch) => {
```
Comment (brusherru) [inline app/redux/network/actions.ts]: Check it out: https://redux-toolkit.js.org/api/createAsyncThunk

**Review by brusherru — COMMENTED** [review]
Welcome and thanks for the contribution!
PR seems good to me, but I think it might be improved a little bit to become a new standard in our project.
Could you?

File: shared/fp-tools.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const compose = <A extends unknown, F extends (arg0: A) => A>(...fns: F[]) => (initialVal: A) => fns.reduce((acc, fn) => fn(acc), initialVal);
```
Comment (maparr) [inline shared/fp-tools.ts]: we have Ramda lib , and compose fn already[ in it ](https://ramdajs.com/docs/#compose)

File: app/redux/createIpcAction.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  onFulfilled: (state, action) => ({ ...state, ...action.payload }),
+  onPending: (state) => state,
+  onRejected: (state, action) => ({ ...state, error: action.payload }),
+};
+
+const createIpcAction = <T>(
+  typePrefix: string,
+  payloadCreator: AsyncThunkPayloadCreator<T>,
+  { onFulfilled, onRejected, onPending } = DEFAULT_IPC_REDUCERS
+): [AsyncThunk<T, void, EmptyObject>, (builder: ActionReducerMapBuilder<T>) => ActionReducerMapBuilder<T>] => {
```
Comment (maparr) [inline app/redux/createIpcAction.ts]: the return type is possible to separate. Not unique to the specific createIpcAction function types, they are universal. May u just place it separately here or in a shared folder for the redux.

File: app/redux/network/actions.ts (hunk 1 above)
Comment (vshakun) [reply app/redux/network/actions.ts]: Thank you!
It helped!

File: shared/fp-tools.ts (hunk 2 above)
Comment (vshakun) [reply shared/fp-tools.ts]: Thank you!
I have tried to use it. Can you review it again, please?

**Review by brusherru — DISMISSED** [review]
LGTM

**Review by brusherru — APPROVED** [review]
LGTM

**Review by maparr — APPROVED** [review]
LGTM!

**Pull request merged**